x-pack/plugins/lists/server/services/exception_lists/exception_list_client.ts

```diff
@@ -90,6 +93,24 @@ export class ExceptionListClient {
     });
   };
 
+  /**
+   * Create the Trusted Apps Agnostic list if it does not yet exist (`null` is returned if it does exist)
+   */
+  public createTrustedAppsList = async (): Promise<ExceptionListSchema | null> => {
+    if (this.trustedAppsListCreated) {
+      return null;
+    }
+    const { savedObjectsClient, user } = this;
+    return createEndpointTrustedAppsList({
+      savedObjectsClient,
+      user,
+      version: 1,
+    }).then((response) => {
+      this.trustedAppsListCreated = true;
+      return response;
+    });
+  };
```

Comment:
Most parts of Kibana core and people in general use `await` with `async and then try/catch with their errors. I don't see many people championing using promises fwiw. Not saying outside of here people don't just mentioning it from a statistics point of view.

If you leave your code as is, it could get converted later because vs-code added an auto suggest for promises to become async/await (regardless of if you have linter rules on or off) but do not auto-suggest it to turn them back into promises.

Examples in my IDE from other code sample:
<img width="911" alt="Screen Shot 2020-07-31 at 10 41 28 AM" src="https://user-images.githubusercontent.com/1151048/90077614-da722780-dcbf-11ea-9bed-8cd6fbcd3afe.png">

More info:
https://dev.to/ben/visual-studio-code-can-now-convert-your-long-chains-of-promisethens-into-asyncawait-automagically-1b1b

Just some perspective if you do enjoy promises over async/await and you're wondering why code parts keep looking like they are being refactored changed into async/await, it usually because someone is doing a "fix all" in their IDE like vs-code and it's auto-changing them.

I used to use promises all the time, but whenever I saw Kibana core and other plugins using async/await everywhere and my own IDE trying to tell me it was a suggested fix I kind of moved over.

Reply:
I'm not a VSCode user, but Webstorm has similar things (I think I turned that one off).

I have for the most part also converted over to using mostly `await`, but there are still cases where my brain reverts back to Promises. This will be refactored as part of removing class state.

Thanks 👍